### ISSUE TITLE
Unable to build pljava on Windows using mingw/msys

### ISSUE THREAD

**Issue by amCap1712** [tada/pljava#282]
I have followed all steps mentions in general building steps of PL/Java but am unable to build it using mingw/msys. The relevant log file is available at https://usercontent.irccloud-cdn.com/file/lhvxuJfO/log.txt

**Comment by jcflack** [comment]
Hi,

Is there a link to the current environment setup that you are using to build?

I notice there are a lot of messages in the log pertaining to an `int64` data type.

Could the issue be something as simple as having `Mingw` set up rather than `Mingw-w64`?

**Comment by jcflack** [comment]
Can you run Maven with `-X` to show more information on exactly what it is running in the build?

**Comment by amCap1712** [comment]
Hi! This is the [log with -X.](https://usercontent.irccloud-cdn.com/file/VXzp0Yd3/new_log.txt)
I also tried to check for the int64 issue and found [this](http://mingw.5.n7.nabble.com/Problem-with-int64-type-td14269.html). Do you think this may relevant ? I tried setting CFLAGS and CPPFLAGS environment variables as mentioned but to no avail.

**Comment by jcflack** [comment]
I think there must be a copy of PostgreSQL built by MSVC present in the appveyor image somewhere, and it is the one the default `pg_config` command is finding. I noticed this line in the `-X` output:

```
[loadresource] #define PG_VERSION_STR "PostgreSQL 12.1, compiled by Visual C++ build " CppAsString2(_MSC_VER) ", 64-bit"
```

The `mvn` command line can be given a `-Dpgsql.pgconfig=` option pointing to the right `pg_config` binary for the PostgreSQL installation to build against, which should be one that was built with Mingw-w64.

There seems to be a PG 12.2 prebuilt package [in pacman][pacman].

[pacman]: https://packages.msys2.org/package/mingw-w64-x86_64-postgresql

**Comment by amCap1712** [comment]
Using the mingw package for PostgreSQL fixed the int64 issues. However, there is another issue. The relevant part of the stack trace is this.
```
Starting Core 0 with 53 source files...
[ERROR] C:\projects\pljava\pljava-so\src\main\c\PgSavepoint.c:75:13: error: conflicting types for 'unwind'
[ERROR]    75 | static void unwind(void (*f)(void), jint xid, jint nestingLevel)
[ERROR]       |             ^~~~~~
[ERROR] C:\projects\pljava\pljava-so\src\main\c\PgSavepoint.c:30:13: note: previous declaration of 'unwind' was here
[ERROR]    30 | static void unwind(void (*f)(void), int nestingLevel, int xid);
[ERROR]       |             ^~~~~~
[ERROR] C:\projects\pljava\pljava-so\src\main\c\SubXactListener.c: In function 'subXactCB':
[WARNING] C:\projects\pljava\pljava-so\src\main\c\SubXactListener.c:45:2: warning: enumeration value 'SUBXACT_EVENT_PRE_COMMIT_SUB' not handled in switch [-Wswitch]
[ERROR]    45 |  switch(event)
[ERROR]       |  ^~~~~~
[ERROR] C:\projects\pljava\pljava-so\src\main\c\XactListener.c: In function 'xactCB':
[WARNING] C:\projects\pljava\pljava-so\src\main\c\XactListener.c:26:2: warning: enumeration value 'XACT_EVENT_PARALLEL_COMMIT' not handled in switch [-Wswitch]
[ERROR]    26 |  switch(event)
[ERROR]       |  ^~~~~~
[WARNING] C:\projects\pljava\pljava-so\src\main\c\XactListener.c:26:2: warning: enumeration value 'XACT_EVENT_PARALLEL_ABORT' not handled in switch [-Wswitch]
[WARNING] C:\projects\pljava\pljava-so\src\main\c\XactListener.c:26:2: warning: enumeration value 'XACT_EVENT_PRE_COMMIT' not handled in switch [-Wswitch]
[WARNING] C:\projects\pljava\pljava-so\src\main\c\XactListener.c:26:2: warning: enumeration value 'XACT_EVENT_PARALLEL_PRE_COMMIT' not handled in switch [-Wswitch]
[WARNING] C:\projects\pljava\pljava-so\src\main\c\XactListener.c:26:2: warning: enumeration value 'XACT_EVENT_PRE_PREPARE' not handled in switch [-Wswitch]
```
Here is the [complete log](https://usercontent.irccloud-cdn.com/file/QshplgGn/log.txt) for your reference.

**Comment by jcflack** [comment]
Ok, that's rather embarrassing.

Can you try the build from branch `bug/master/issue282`?

I think that'll fix it, but would love to see CI confirmation before merging. :)

**Comment by amCap1712** [comment]
Yes, the issue is fixed. The build now works on mingw-w64.

**Comment by jcflack** [comment]
Believed resolved in 1.5.6.
